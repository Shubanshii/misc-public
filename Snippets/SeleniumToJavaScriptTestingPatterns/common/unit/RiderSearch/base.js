import React from 'react'
import PageObjectBasePage from '../../../_patterns/page-object/poBasePage.pattern'

import pageElemMap from './elementmap'
import RiderSearchComponent from '../../../../client/components/rider-search/RiderSearch'


import { riderSearchProps } from '../../../_mocks/common/RiderSearch'

export class RiderSearch extends PageObjectBasePage
{
  constructor(validatorType,renderMethod='mount')
  {
    super({
      component:<RiderSearchComponent {...riderSearchProps} />,
      pageElemMap,
      pageValidator:validatorType,
    },renderMethod);
  }
}
