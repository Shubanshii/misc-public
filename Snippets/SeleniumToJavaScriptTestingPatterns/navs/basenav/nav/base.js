import React from 'react'
import { Provider } from 'react-redux'
import configureStore from '../../../../client/store/configureStore.dev'

import propsSingleton from '../../../_patterns/singleton/singletonBase.pattern'

import pageElemMap from './elementmap'
import PageObjectBasePage from '../../../_patterns/page-object/poBasePage.pattern'
import {sinonNavActions, sinonNavProps} from './sinonBase'

import Footer from '../../../../client/components/common/Footer'
import Header from '../../../../client/components/common/Header'
import RiderSearch from '../../../../client/components/rider-search/RiderSearch'
import VerticalNav from '../../../../client/components/common/VerticalNav'

import AlertCards from '../../../../client/components/alerts/AlertCards'


import { topNavProps,sidebarProps,footerProps,alertCardsProps } from '../../../_mocks/navs/basenav'

export class TopNav extends PageObjectBasePage
{
  constructor(validatorType,renderMethod='mount')
  {
    super({
      component:<Header {...topNavProps} changeRoute={(param)=>{console.log('changing route',param);}} />,
      pageElemMap,
      pageValidator:validatorType,
    },renderMethod);
  }
}

export class SideBar extends PageObjectBasePage
{
  constructor(validatorType)
  {
    super({
      component:<VerticalNav {...sidebarProps}/>,
      pageElemMap,
      pageValidator:validatorType
    },'render');
  }
}
export class SideBarPII extends PageObjectBasePage
{
  constructor(validatorType)
  {
    super({
      component:<VerticalNav {...sidebarProps} user={{...sidebarProps.user,user_type:'PII'}} />,
      pageElemMap,
      pageValidator:validatorType
    },'render');
  }
}

export class FooterNav extends PageObjectBasePage
{
  constructor(validatorType)
  {
    super({
      component:<Footer {...footerProps} />,
      pageElemMap,
      pageValidator:validatorType
    },'render');
  }
}

export class FooterNavNoReload extends PageObjectBasePage
{
  constructor(validatorType)
  {
    super({
      component:<Footer {...footerProps} user={{...footerProps.user,email:'no_reload'}} />,
      pageElemMap,
      pageValidator:validatorType
    },'render');
  }
}

export class Alerts extends PageObjectBasePage
{
  constructor(validatorType,renderMethod='mount')
  {
    const alertsProps = propsSingleton.get('fullPageData')

    alertsProps.actions = { ...sinonNavActions } // <-- to add to nested props

    super({
      component:<AlertCards {...alertsProps} {...sinonNavProps} />,
      pageElemMap,
      pageValidator:validatorType
    },renderMethod);
  }
}
