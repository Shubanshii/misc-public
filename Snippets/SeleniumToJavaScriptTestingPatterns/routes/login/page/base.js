import React from 'react'
import { Provider } from 'react-redux'
import configureStore from '../../../../client/store/configureStore.dev';

import PageObjectBasePage from '../../../_patterns/page-object/poBasePage.pattern'

import pageElemMap from './elementmap'
import pageValidator from './validator'

import AuthForm from '../../../../client/components/user-auth/AuthForm';
import { authProps } from '../../../_mocks/pages/authroute/props';

import { inputChangeHelper } from '../../../_utils/enzyme.utils';

export default class LoginBase extends PageObjectBasePage
{
  constructor()
  {
    super({
      component:<AuthForm {...authProps} />,
      pageElemMap,
      pageValidator
    });
  }

  formFocus()
  { this.Map.inputUser.simulate('focus'); }
  changeInputVal(o)
  {
    this.Map.inputUser.simulate('change', inputChangeHelper(o.username));
    this.Map.inputPassword.simulate('change',  inputChangeHelper(o.pass) );
  }
  submitForm()
  { this.Map.loginSubmitButton.simulate('click') }
}
