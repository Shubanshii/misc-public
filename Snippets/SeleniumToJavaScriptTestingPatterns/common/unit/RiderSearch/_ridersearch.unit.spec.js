import unitValidator from './validator'

import { RiderSearch } from './base'

describe('Rider Search Common Component (Unit)', () => {
  let componentBase;

  beforeEach(() => componentBase = new RiderSearch(unitValidator,'render'));

  test('Rider Search has input field', ()=>{
    componentBase.Validate().hasInputField();
    componentBase.Validate().hasPlaceholderText('Search by Rider Name, DOB, Rider or Ride ID#, Phone or Address')
  })

  test('Layout is unchanged', () =>
  { componentBase.Snap() })

});
