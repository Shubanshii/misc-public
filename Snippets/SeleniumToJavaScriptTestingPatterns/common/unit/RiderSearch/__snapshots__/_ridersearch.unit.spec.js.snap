// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rider Search Common Component (Unit) Layout is unchanged 1`] = `
<div
  className="home"
>
  <div
    className="flex-wrapper"
  >
    <h1
      className="center-text"
    >
      Rider Directory
    </h1>
    <form
      className="rider-search relative-position form-column "
      onKeyDown={[Function]}
    >
      <div
        className="input-wrapper two-column-input-wrapper"
      >
        <input
          autoComplete="off"
          className="bypass-read-only "
          data-valid={true}
          icon="search"
          id="rider-search-home"
          label="Search by Rider Name, DOB, Rider or Ride ID#, Phone or Address"
          name="riderSearch"
          onChange={[Function]}
          onFocus={[Function]}
          tabIndex={1}
          type="text"
          value=""
        />
        <label
          htmlFor="rider-search-home"
        >
          <span
            className="icon-form"
          >
            <svg
              className=""
              onClick={null}
            >
              <svg
                id="icon-search"
                viewBox="0 0 35 35"
              >
                <title>
                  search
                </title>
                <path
                  d="M31.008 27.231l-7.58-6.447c-0.784-0.705-1.622-1.029-2.299-0.998 1.789-2.096 2.87-4.815 2.87-7.787 0-6.627-5.373-12-12-12s-12 5.373-12 12 5.373 12 12 12c2.972 0 5.691-1.081 7.787-2.87-0.031 0.677 0.293 1.515 0.998 2.299l6.447 7.58c1.104 1.226 2.907 1.33 4.007 0.23s0.997-2.903-0.23-4.007zM12 20c-4.418 0-8-3.582-8-8s3.582-8 8-8 8 3.582 8 8-3.582 8-8 8z"
                />
              </svg>
            </svg>
          </span>
          <span
            className="label-content"
          >
            Search by Rider Name, DOB, Rider or Ride ID#, Phone or Address
          </span>
        </label>
        <span
          className="validation"
        >
          
        </span>
      </div>
      <div>
        <ul
          className="search-results home-search-results "
          id="resultList"
        >
          
        </ul>
      </div>
    </form>
  </div>
</div>
`;
